File: 프로그래머스/unrated/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.java
class Solution {
    public int solution(String myString, String pat) {
        int answer = 0;
        String str = "";

        for (int i = 0; i < myString.length(); i++) {
            if (myString.charAt(i) == 'A') {
                str += "B";
            } else {
                str += "A";
            }
        }

        if (str.contains(pat)) {
            answer = 1;
        } else {
            answer = 0;
        }
        
        return answer;
    }
}

File: 프로그래머스/0/181864. 문자열 바꿔서 찾기/문자열 바꿔서 찾기.py
def solution(myString, pat):
    tmp_list = []
    for str in myString:
        if str == 'A':
            tmp_list.append('B')
        else:
            tmp_list.append('A')

    result = ''.join(tmp_list)

    if pat in result:
        return 1
    else:
        return 0